app/__init__.py

```diff
@@ -1,169 +1,197 @@
-
-        if request_data[0]["user_id"] == user_id:
-            if len(request_data) != 0:
-                return jsonify({"request": request_data[0]}), 200
-    return jsonify({"message": "Sign In to view requests"}), 403
+@jwt_required
+def user_get_request(request_id):
+    user_id = get_jwt_identity()
+    user_request = get_user_request(user_id, request_id)
+    if len(user_request) == 0:
+        return jsonify({
+            "message": "Request Not Found"
+        }), 403
+    return jsonify({"request": user_request}), 200
 
 
 # UPDATE(PUT) a request
 @app.route('/api/v1/users/requests/<request_id>', methods=['PUT'])
-def update_request(request_id):
-    if len(session) != 0:
-        user_id = session["user_id"]
-        request_data = []
-        for r in requests:
-            if r["id"] == request_id:
-                request_data.append(r)
-        if request_data[0]["user_id"] == user_id:
-            if len(request_data) != 0:
-                request_data[0]["title"] = request.json.get(
-                    "title", request_data[0]["title"])
-                request_data[0]["type"] = request.json.get(
-                    "type", request_data[0]["type"])
-                request_data[0]["description"] = request.json.get(
-                    "description", request_data[0]["description"])
-                request_data[0]["category"] = request.json.get(
-                    "category", request_data[0]["category"])
-                request_data[0]["area"] = request.json.get(
-                    "area", request_data[0]["area"])
-
-                return jsonify({"requests": request_data[0]}), 200
-            return jsonify({"message": "Request Not Found"}), 404
-        return jsonify({"message": "Cannot Access Request"}), 403
-    return jsonify({"message": "Sign In to view requests"}), 403
+@jwt_required
+def user_update_request(request_id):
+    user_id = get_jwt_identity()
+    user_request = get_user_request(request_id, user_id)
+
+    # Ensure input is in json format
+    if not request.is_json:
+        return jsonify({"msg": "Invalid data format"}), 400
+    for key in request.json:
+        # Ensure input contains data
+        if request.json[key] == "":
+            return jsonify({"message": "All Fields Required"}), 400
+        # Ensure input data is a string
+        elif type(request.json[key]) != str:
+            return jsonify({"message": "Input Must be a String"}), 400
+
+    # Ensure request has been fetched
+    if len(user_request) != 0:
+        update_request(request_id,
+                       user_id,
+                       request.json.get("title").lower(),
+                       request.json.get("description").lower(),
+                       request.json.get("_type").lower(),
+                       request.json.get("category").lower(),
+                       request.json.get("area").lower()
+                       )
+        return jsonify({"Success": user_request}), 200
```

Comment:
remember the specifications are that a user should not be able to edit a request that has already been approved by the admin.